src/components/app/ErrorBoundary.js

```diff
@@ -0,0 +1,85 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+// @flow
+
+import * as React from 'react';
+import { reportError } from '../../utils/analytics';
+import './ErrorBoundary.css';
+
+type State = {|
+  hasError: boolean,
+  showDetails: boolean,
+  error?: Error,
+  componentStack?: string,
+|};
+
+type Props = {|
+  +children: React.Node,
+  +message: string,
+|};
+
+/**
+ * This component will catch errors in components, and display a more friendly error
+ * message. This stops the entire app from unmounting and displaying an empty screen.
+ *
+ * See: https://reactjs.org/docs/error-boundaries.html
+ */
+export class ErrorBoundary extends React.Component<Props, State> {
+  state = {
+    hasError: false,
+    showDetails: false,
+  };
+
+  componentDidCatch(
+    error: mixed,
+    { componentStack }: { componentStack: string }
+  ) {
+    console.error(
+      'An unhandled error was thrown in a React component.',
+      error,
+      componentStack
+    );
+    this.setState({ hasError: true, componentStack });
```

Comment:
you miss the actual error here, and as a result it's missing in the rendering:

![image](https://user-images.githubusercontent.com/454175/51680932-6a1f8a80-1fe3-11e9-81ad-8a29aaf4beb6.png)